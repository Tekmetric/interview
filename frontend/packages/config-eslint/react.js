module.exports = {
  extends: [
    '@vercel/style-guide/eslint/browser',
    '@vercel/style-guide/eslint/typescript',
    '@vercel/style-guide/eslint/react'
  ].map(require.resolve),
  parserOptions: {
    project: '**/tsconfig.json',
    sourceType: 'module'
  },
  globals: {
    JSX: true
  },
  settings: {
    'import/resolver': {
      typescript: {
        project: '**/tsconfig.json',
        sourceType: 'module'
      },
      node: {
        extensions: ['.mjs', '.js', '.jsx', '.ts', '.tsx']
      }
    }
  },
  ignorePatterns: ['node_modules/', 'dist/', '.eslintrc.js', '**/*.css'],
  // add rules configurations here
  rules: {
    'import/no-default-export': 'off',
    'import/no-relative-packages': 'off',
    'react/button-has-type': 'off',
    'no-restricted-imports': [
      'error',
      {
        patterns: [
          {
            group: ['packages/*'],
            message: "Please use '@tekmetric/*' imports instead"
          }
        ]
      }
    ],
    'import/order': [
      'error',
      {
        'newlines-between': 'always',
        groups: [
          ['builtin', 'external'],
          'unknown',
          ['parent', 'sibling', 'index']
        ],
        pathGroups: [
          {
            pattern: '~/**',
            group: 'unknown',
            position: 'after'
          }
        ]
      }
    ],
    'react/function-component-definition': [
      'error',
      {
        namedComponents: 'arrow-function',
        unnamedComponents: 'arrow-function'
      }
    ]
  },
  overrides: [
    {
      files: ['*.config.js'],
      env: {
        node: true
      }
    },
    {
      files: ['*.ts', '*.tsx'],
      rules: {
        'no-unused-vars': 'off',
        '@typescript-eslint/no-invalid-void-type': 'off',
        'react/jsx-no-leaked-render': 'off',
        '@typescript-eslint/no-unused-vars': [
          'error',
          {
            vars: 'all',
            args: 'after-used',
            ignoreRestSiblings: true,
            argsIgnorePattern: '^_',
            varsIgnorePattern: '^_'
          }
        ],
        '@typescript-eslint/no-misused-promises': [
          'error',
          {
            checksVoidReturn: false
          }
        ]
      }
    },
    {
      files: ['*.test.ts', '*.test.tsx'],
      env: {
        jest: true
      },
      rules: {
        '@typescript-eslint/no-unsafe-call': 'off',
        '@typescript-eslint/no-unsafe-member-access': 'off'
      }
    }
  ]
}
