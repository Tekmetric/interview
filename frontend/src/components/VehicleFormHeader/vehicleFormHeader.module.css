.header {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.text {
  margin-bottom: var(--space-md);
}

.text a svg {
  margin-right: var(--space-xs);
}

.text a,
.text button {
  margin: var(--space-sm) 0;
  align-items: center;
  justify-content: center;
  display: flex;
  width: 100%;
}

.buttons {
  display: flex;
  flex-direction: column;
}

.buttons .active {
  background-color: var(--primary-color);
}

.image {
  position: relative;
}

.image button {
  position: absolute;
  top: 5px;
  right: 5px;
}

.image img {
  object-fit: cover;
  width: 150px;
  height: 100px;
  border-radius: var(--border-radius-md);
  border: 1px solid #fff;
}

@media (min-width: 768px) {
  .header {
    flex-direction: row;
    margin-bottom: var(--space-md);
  }

  .text a,
  .text button {
    width: 125px;
  }
  .text {
    margin-right: var(--space-md);
  }
  .buttons {
    column-gap: 1rem;
    display: flex;
    flex-direction: row;
  }
}

@media (min-width: 1024px) {
  .image img {
    width: 300px;
    height: 200px;
  }
}
