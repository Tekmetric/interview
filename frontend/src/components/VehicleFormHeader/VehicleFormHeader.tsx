import { FC } from 'react';
import { Link } from 'react-router-dom';
import { FaArrowLeft, FaTrash } from 'react-icons/fa';
import styles from './vehicleFormHeader.module.css';
import { VehicleFormHeaderProps } from '../../features/vehicles/types';

const VehicleFormHeader: FC<VehicleFormHeaderProps> = ({
  isEditMode,
  isDisabled,
  onEditClick,
  vehicleTitle,
  imageUrl,
  onDeleteImage,
}) => (
  <div className={styles.header}>
    <div className={styles.text}>
      <div className={styles.buttons}>
        <Link className="link-button" to={`/`}>
          <FaArrowLeft />
          <span>Vehicles</span>
        </Link>
        {isEditMode && (
          <button type="button" className={isDisabled ? styles.active : ''} onClick={onEditClick}>
            Edit
          </button>
        )}
      </div>
      <h1>{vehicleTitle}</h1>
    </div>
    <div className={styles.image}>
      {imageUrl && !isDisabled && (
        <button onClick={onDeleteImage} aria-label="delete image">
          <FaTrash aria-hidden="true" />
        </button>
      )}
      <img src={imageUrl || 'https://placehold.co/400'} alt="Vehicle Preview" />
    </div>
  </div>
);

export default VehicleFormHeader;
