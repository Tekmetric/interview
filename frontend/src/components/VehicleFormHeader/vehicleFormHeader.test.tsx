import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import VehicleFormHeader from './VehicleFormHeader';

describe('VehicleFormHeader', () => {
  const mockOnDeleteImage = vi.fn();
  const mockOnEditClick = vi.fn();

  it('renders correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <VehicleFormHeader
          imageUrl={null}
          isDisabled={true}
          isEditMode={false}
          onDeleteImage={mockOnDeleteImage}
          onEditClick={mockOnEditClick}
          vehicleTitle={'Honda CRV, 2018'}
        />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
