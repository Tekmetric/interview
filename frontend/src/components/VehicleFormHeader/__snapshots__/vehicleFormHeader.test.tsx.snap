// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VehicleFormHeader > renders correctly 1`] = `
<div>
  <div
    class="_header_801691"
  >
    <div
      class="_text_801691"
    >
      <div
        class="_buttons_801691"
      >
        <a
          class="link-button"
          data-discover="true"
          href="/"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
            />
          </svg>
          <span>
            Vehicles
          </span>
        </a>
      </div>
      <h1>
        Honda CRV, 2018
      </h1>
    </div>
    <div
      class="_image_801691"
    >
      <img
        alt="Vehicle Preview"
        src="https://placehold.co/400"
      />
    </div>
  </div>
</div>
`;
